biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -418,6 +433,237 @@ end subroutine get_harvest_rate_area
+                 harvestable_forest_c(h_index) = harvestable_forest_c(h_index) + harvestable_patch_c
+              end if
+           else if (currentPatch%anthro_disturbance_label .eq. secondaryforest .and. &
+                currentPatch%age_since_anthro_disturbance < secondary_age_threshold) then
+              ! Secondary young
+              if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH2") then
+                 harvestable_forest_c(h_index) = harvestable_forest_c(h_index) + harvestable_patch_c
+              end if
+           end if
+        end do
+        currentPatch => currentPatch%younger
+     end do
+
+   end subroutine get_harvestable_carbon
+
+   ! ============================================================================
+
+   subroutine get_harvest_rate_carbon (patch_anthro_disturbance_label, hlm_harvest_catnames, &
+                 hlm_harvest_rates, secondary_age, harvestable_forest_c, &
+                 harvest_rate, harvest_tag, cur_harvest_tag)
+
+     ! -------------------------------------------------------------------------------------------
+     !
+     !  DESCRIPTION:
+     !  get the carbon-based harvest rates based on info passed to FATES from the boundary conditions in.
+     !  assumes logging_time == true
+
+      ! Arguments
+      real(r8), intent(in) :: hlm_harvest_rates(:) ! annual harvest rate per hlm category
+      character(len=64), intent(in) :: hlm_harvest_catnames(:) ! names of hlm harvest categories
+      integer, intent(in) :: patch_anthro_disturbance_label    ! patch level anthro_disturbance_label
+      real(r8), intent(in) :: secondary_age     ! patch level age_since_anthro_disturbance
+      real(r8), intent(in) :: harvestable_forest_c(:)  ! site level forest c matching criteria available for harvest
+      real(r8), intent(out) :: harvest_rate
+      integer,  intent(inout) :: harvest_tag(:)  ! 0. normal harvest; 1. current site does not have enough C but
+                                                 ! can perform harvest by ignoring criteria; 2. current site does
+                                                 ! not have enough carbon
+                                                 ! This harvest tag shall be a patch level variable but since all
+                                                 ! logging functions happen within cohort loop we can only put the 
+                                                 ! calculation here. Can think about optimizing the logging calculation
+                                                 ! in the future.
+      integer,  intent(out), optional :: cur_harvest_tag  ! harvest tag of the current cohort
+
+      ! Local Variables
+      integer :: h_index   ! for looping over harvest categories
+      integer :: icode   ! Integer equivalent of the event code (parameter file only allows reals)
+      real(r8) :: harvest_rate_c    ! Temporary variable
+      real(r8) :: harvest_rate_supply  ! Temporary variable
```

Comment:
what are the units of the above two variables?

Reply:
kgC site-1. Will add.